### PR TITLE
Added Unit Test for WindowsSmtpConfig Plugin

### PR DESCRIPTION
**Pull request by vasilii-b**
### Description

This PR cover `Magento\Email\Test\Unit\Model\Plugin\WindowsSmtpConfig` plugin with unit test.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Email/Test/Unit/Model/Plugin/WindowsSmtpConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
```
Comment (rogyar) [inline app/code/Magento/Email/Test/Unit/Model/Plugin/WindowsSmtpConfigTest.php:7]: Please, declare the strict types.

File: app/code/Magento/Email/Test/Unit/Model/Plugin/WindowsSmtpConfigTest.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    public function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+
+        $this->osInfoMock = $this->createMock(OsInfo::class);
+        $this->configMock = $this->createMock(ReinitableConfigInterface::class);
+        $this->transportMock = $this->createMock(TransportInterface::class);
+
+        $this->windowsSmtpConfigMock = $objectManager->getObject(
```
Comment (rogyar) [inline app/code/Magento/Email/Test/Unit/Model/Plugin/WindowsSmtpConfigTest.php]: It does not look like a "mock". We test the real object. Please, fix the naming.

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
Hello @vasilii-b. Thank you for your contribution. Please, consider fixing a few minor issues mentioned in the review

**Comment by vasilii-b** [comment]
Thank you @rogyar, will do.

**Comment by vasilii-b** [comment]
Hi @rogyar , I have added requested changes, but I see the Codacy failed with a strange issue. Could you please have a look and advice what to do there ? 🤔  Thank you!

**Comment by rogyar** [comment]
I've double checked the result and see no issue. So, no worries

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-3234](https://jira.corp.magento.com/browse/ENGCOM-3234) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**